core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -462,6 +462,9 @@ private[spark] object SparkHadoopUtil {
     for ((key, value) <- conf.getAll if key.startsWith("spark.hadoop.")) {
       hadoopConf.set(key.substring("spark.hadoop.".length), value)
     }
+    if (conf.getOption("spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version").isEmpty) {
+      hadoopConf.set("mapreduce.fileoutputcommitter.algorithm.version", "1")
```

Comment:
What about we just use `hadoopConf.setIfUnset`?

Reply:
Thanks for review, but we cannot use it because Apache Hadoop already loads the default value of `mapreduce.fileoutputcommitter.algorithm.version`, @HyukjinKwon .

Reply:
Gotya